zerver/lib/actions.py

```diff
@@ -557,7 +562,9 @@ def revoke_preregistration_users(
 
     if used_preregistration_user is not None:
         used_preregistration_user.status = confirmation_settings.STATUS_ACTIVE
-        used_preregistration_user.save(update_fields=["status"])
+        used_preregistration_user.created_user_profile = created_user_profile
+
+        used_preregistration_user.save(update_fields=["status", "created_user_profile"])
```

Comment:
This seems like a clean enough place to do this, but we probably need to rename  `revoke_preregistration_users` to `process_user_preregistration_user` so the name is no longer misleading?

Reply:
I renamed it to `process_preregistration_users` as it processes multiple preregistration users at the same time. Added a docstring to describe its behavior.